src/Orleans/Serialization/OrleansJsonSerializer.cs

```diff
@@ -1,27 +1,23 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
 using System.Net;
-using System.Runtime.Serialization.Formatters.Binary;
-using System.Text;
-using System.Threading.Tasks;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 using Orleans.Runtime;
 
 namespace Orleans.Serialization
 {
-    internal class OrleansJsonSerializer : IExternalSerializer
+    public class OrleansJsonSerializer : IExternalSerializer
     {
-        public static JsonSerializerSettings settings;
+        public static JsonSerializerSettings SerializerSettings { get; set; }
```

Comment:
We can left it public and add Orleans converters on each set. But the this settings were adjusted for the internal usage, and already been used in azure provider. So maybe it should be made private, and users will be provided with different API, so the fallback serializer settings will be left intact.